player/screenshot.c

```diff
@@ -482,22 +483,28 @@ void cmd_screenshot(void *p)
         }
     }
 
-    cmd->success = false;
-
     struct image_writer_opts *opts = mpctx->opts->screenshot_image_opts;
     bool high_depth = image_writer_high_depth(opts);
 
-    struct mp_image *image = screenshot_get(mpctx, mode, high_depth);
+    cmd->success = false;
 
-    if (image) {
-        char *filename = gen_fname(cmd, image_writer_file_ext(opts));
-        if (filename)
-            cmd->success = write_screenshot(cmd, image, filename, NULL);
-        talloc_free(filename);
-    } else {
+    struct mp_image *image = screenshot_get(mpctx, mode, high_depth);
+    if (!image) {
         mp_cmd_msg(cmd, MSGL_ERR, "Taking screenshot failed.");
+        return;
     }
-
+    
+    char *filename = gen_fname(cmd, image_writer_file_ext(opts));
```

Comment:
Not really sure why you're rearranging all of this? If all you want to do is add some stuff to a node, you can accomplish this by just adding it under the `if (filename)` condition that's in the original code.

Reply:
Well true... But at least now there isn't so much nesting. I can revert that if you want it though.

Reply:
I would prefer the code changes to be only what's actually needed.

Reply:
Done. I'll squash on merge as to not pollute the log.